src/utils/date/date.js

```diff
@@ -124,3 +124,80 @@ const getTimezoneOffsetInMilliseconds = date => {
   const timezoneOffset = date.getTimezoneOffset()
   return timezoneOffset * MILLISECONDS_IN_A_MINUTE
 }
+
+export const getDatesOfTheWeekEnd = date => {
+  const dayOfTheWeek = date.getDay()
+  if (dayOfTheWeek === SUNDAY_INDEX_IN_A_WEEK) {
+    return [date]
+  }
+  const daysUntilSaturday = SATURDAY_INDEX_IN_A_WEEK - dayOfTheWeek
+  const timestampOfNextSaturday = date.getTime() + MILLISECONDS_IN_A_DAY * daysUntilSaturday
+  const dateOfNextSaturday = new Date(timestampOfNextSaturday)
+
+  const daysUntilSunday = DAYS_IN_A_WEEK - dayOfTheWeek
+  const timestampOfNextSunday = date.getTime() + MILLISECONDS_IN_A_DAY * daysUntilSunday
+  const dateOfNextSunday = new Date(timestampOfNextSunday)
+
+  return [dateOfNextSaturday, dateOfNextSunday]
+}
+
+export const getBeginningAndEndingDatesForGivenTimeRange = (date, timeRange) => {
+  const beginningDatetime = getDateAtGivenTime(date, timeRange[0])
+  const endingDatetime = getDateAtGivenTime(date, timeRange[1])
+  return [beginningDatetime, endingDatetime]
+}
+
+export const getBeginningAndEndingTimestampsForGivenTimeRange = (date, timeRange) => {
+  return getBeginningAndEndingDatesForGivenTimeRange(date, timeRange).map(date =>
+    getTimestampFromDate(date)
+  )
+}
+
+const getDateAtGivenTime = (date, time) => {
+  if (time === 24) {
```

Comment:
```suggestion
  if (time === 24) {
```
Exporter dans une constante plus parlante ?

Reply:
Vu ensemble, on a même pu supprimer cette condition pour mon plus grand bonheur